pkg/ipamd/ipamd.go

```diff
@@ -447,7 +447,7 @@ func (c *IPAMContext) nodeInit() error {
 	increasedPool, err := c.tryAssignIPs()
 	if err == nil && increasedPool {
 		c.updateLastNodeIPPoolAction()
-	} else {
+	} else if err != nil {
 		return err
 	}
```

Comment:
styling nit: instead of checking both err == nil and err != nill

if err != nil { return err } 
else if increasedPool {..}